e2e_tests/tests/cluster/test_checkpoints.py

```diff
@@ -91,6 +92,7 @@ def run_gc_checkpoints_test(checkpoint_storage: Dict[str, str]) -> None:
                     elif checkpoint["state"] == "DELETED":
                         try:
                             with storage_manager.restore_path(storage_id):
+                                exp.print_trial_logs(trial_id)
                                 raise AssertionError("checkpoint not deleted")
```

Comment:
this strategy, which is basically 'bail on the first failure' is quick to write but gives very little information when it actually fails.

Instead of 'bail on first failure', can you make this a 'gather a list of failures and raise an exception with all failures'?